### DISCUSSION TITLE
2023-07-14 - Data Engineering Review - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#6002] [Chewing the Fat]
Hey data engineers!

After watching the [Chewing the Fat review](https://youtu.be/qkkan1eJJkk)... 

Do you have any comments? (Think about what you have learned throughout the data engineering series)

**Comment by GordonBeeming** [comment]
Retros are great, in some ways this is a retro on the series where we can look back and see the impact that it has had 😁. I've enjoyed the series and the follow-up and digesting the fats fat is great too.

**Reply by PennyWalker** [reply]
Agreed, retros are where we learn the most and I got a lot out of seeing the trends and interesting comments. 

This was a great learning series that built upon itself each week nicely!

**Comment by jakebayliss** [comment]
This was a great series and a very important topic for us. Everyone should know at least the fundamentals of data modelling!
Would love to see us put some of these concepts together in our own organisation

**Comment by leahy268** [comment]
Loved seeing the results of our Chewing the Fat.  Retros are how we learn. Thoroughly enjoyed this whole series as I've often been asked to find the answers to questions using data.  I haven't always had all the tools at my disposal.

**Comment by ozdemir-mehmet** [comment]
Interesting to see there was a lack of data knowledge (I didn't expect that), but there is an appetite to learn more.

**Comment by kikibianc** [comment]
Interesting to learn more about this topic - while it's not something I touch every day, it's definitely an important concept to nail.

**Comment by Marxoz** [comment]
I was surprised to find how many of our devs are not familiar with data, in particular, how data can/should be stored, and how to query that data. 

These chewing the facts have been gold for everyone to understand what happens behind the scenes of API's that they probably consume on a daily basis.

It is also important to understand the data of a client to identify opportunities for SSW to help them clean and report on that data.

**Comment by MarcusIrmscherSSW** [comment]
As a non Dev the intricacies of Data Engineering are lost on me, but I know have a better understanding of the importance of good Data Engineering. 

I've learnt that Data engineering is crucial as it ensures the smooth and efficient management of data to make it usable for analysis and decision-making.

**Comment by 2fernandez** [comment]


**Reply by jaydenalchin** [reply]
+1 to that!

**Comment by UlyssesMaclaren** [comment]
What Penny said... Account Managers need to seem knowledgeable about at least the broad strokes of what we're selling.

**Comment by lukecookssw** [comment]
It was great to have more than one 15 minute window to explore a topic - especially something as broad and complex as data engineering. Happy to see that most people got a lot out of it.

**Comment by cvionnet** [comment]
It's great to see the results of CTF.
The series was super helpful to have a better understanding of the different ways of storing and organizing data. It also helps to demystify a lot of concepts.

**Comment by vladislav-kir** [comment]
I loved learning more about data engineering through this series. It was also interesting to watch digesting the fat!

**Comment by bahjatalaadelssw** [comment]
It's an important topic. I'd prefer a distilled edit to only highlight the insights. I understand it's more pre-production, but the end videos would be more watchable.

**Reply by jomelsontanco** [reply]
A "talk-show roundtable" approach seems to be a better format for retrospectives such as this. Although, as you said it might be better to have a summary video when it is done for a CTF. Having two versions of the same video looks to be a good compromise. A longer video version for folks looking for more details and a shorter video for the CTF might work well.

**Comment by jimmy1992abc** [comment]
I like this topic. I can see the value that we are not just "filling the form", but making contributions.

**Comment by AntPolkanov** [comment]
Really enjoyed the data series! 
It was also very interesting to see the response analysis. Can we please digest the fat more often? 🙂

**Comment by AshAnil** [comment]
I am happy that now I know what are Databases, Data lakes, Data warehousing, and the difference between SQL and NoSQL. 
The coolest part is the Retro.. looking back and remembering what we learned.

**Comment by zacharykeeping** [comment]
I really liked that this was a series spread over a few weeks to allow for a deeper dive into the individual aspects of the topic. It's also good to see where we're at in terms of knowledge and experience when it comes to data and where we should be upskilling on.

**Comment by CoffeeBugHub** [comment]
The complete set of data is highly beneficial for someone who lack relevant knowledge, as it enables them to gain a comprehensive understanding of utilizing data in the current era of big data.

**Comment by KahaMason** [comment]
Really enjoyed learning more about data engineering. I've lightly touched on Data warehouses in the past but it was great learning about Data Lakes.
The digesting the fat and the breakdown was also interesting to see. 👍

**Comment by Ben0189** [comment]
I have learn various data storage options, data processing techniques, and their applications in different use cases. I like the CTF series format as it allow me to have progressive learning on a specific topic

**Comment by piers-sinclair** [comment]
I think it's a valuable topic, but we've only just scratched the surface layer.

Practically applying and delving into the technical specifics of these solutions would give a greater understanding to our devs.

**Comment by apforsyth** [comment]
Understanding business data and how to extract it's value is a competitive advantage no company can afford to miss. This SSW CTF series is immensely valuable not only from a technical point of view but also for business decisions as we move into the next generation of AI and machine learning driven apps.

It's also valuable for us video people to understand how unstructured data is treated and how it will change video applications into the future - eg evolution of recommendation engines. Who'd have thought understanding data could be so exciting!

Great analysis of the results from Adam and Piers, thanks for the education.

It's also becoming increasingly important for data engineers and organisations to consider the security and ethical ramifications of the solutions they are building.  There are serious risks with how data is used, particularly in politics, financial and social spheres. Cyber criminals now have access to powerful AI tools and it's tough for regulators and law enforcement to keep up.

Here's a scary video shared yesterday by Australia's e-Safety Commissioner, Julie Inman Grant, which is aimed at parents and kids https://www.linkedin.com/posts/ugcPost-7085265892598554624-SIWE?utm_source=share&utm_medium=member_desktop

-Andrew

**Comment by steven0x51** [comment]
Really love the series-based CTF, it's able to provide deeper insight into the topic.

**Comment by jeoffreyfischer** [comment]
I really loved learning about the different types of data storage! It's a topic I never heard before and I feel the video series is very pedagogic and interesting.

**Comment by christoment** [comment]
Data engineering is a really interesting topic, but we only touched the outer layer of it. Keen to actually play around and understanding more 😊

**Comment by Hona** [comment]
Everyone loves data, using it, collecting it & gaining BI from it.

BUT - a lot of devs are not skilled in it! So it was awesome to dip our feet in a bit

**Comment by jackreimers** [comment]
I think the series was a great introduction to data warehouses and lakes for those of us who may have never heard of or used them, and has also helped in seeing who has knowledge on them already if I need some help with them.

**Comment by NadeekaJK** [comment]
Love analyzing stuff and working with data. It was interesting to see the results of the CTFs. Will definitely come in handy with the data driven models in the business world today. This CTF series motivated me to look into data processing tech more.

**Comment by Abf09** [comment]
Great Series, Data is becoming crucial for the growth of any organizations and individuals as well. Being able to leverage Data for better decision making can solve a number of problems such as Client needs, Predictive analytics, sales, Quality assurance and managing risk.

**Comment by JackDevAU** [comment]
Loved the series content, looking forward to more in this format! Keen to see a workshop learning more about this!

**Comment by amankumarrr** [comment]
I thoroughly enjoyed it (Digesting the Fat). It significantly enhanced my interest as I observed the responses from participants and obtained a concise summary of our CTF series, combined with the actual data. It proved to be a valuable opportunity to gain insights and extract meaningful information from our CTFs.

**Comment by wicksipedia** [comment]
It was a nice change. It felt a little too high level - would've loved to have seen something with more detail e.g. we have a client, they need to report on xxx, here's a data lake/warehouse, lets look at it and extract the info we need and report on it - here's how to do it

**Reply by apforsyth** [reply]
Great idea for a user group

**Comment by stephanfakossw** [comment]
Business data is super important for any organisation. Also, I do love the fact that we are doing a retro :-)

**Comment by leehawkinsssw** [comment]
A series gives us the opportunity to cover a topic with more breadth and depth and this topic was well chosen. While I have a long background in databases (having worked most of my career for a database tooling vendor), I realise a lot of devs don't have intimate data-related knowledge so exposing the various important data concepts in a longer series like this is highly valuable.

The break from AI-related CTFs was very welcome, they didn't generally feel relevant or interesting to me.

**Comment by Khaled-Albahsh** [comment]
Learning data engineering is a make it or break it deal. getting more informed on it only makes us stronger developer. cant say no to that

**Comment by tombui99** [comment]
It's interesting to see the results and responses from the video. I would love to learn more about data engineering, even though it's not something I would work with every day, it's definitely an important topic in the near future.

**Comment by JaneYangssw** [comment]
The series content is great. It can effectively guide novices to learn more about data engineering, providing us with ideas for addressing future customer demands.

**Comment by Harry-Ross** [comment]
I think it was great to be able to see how everyone else in the company felt about data, and an aggregate of how skilled other devs were at data in the company. I also think its great that the CTF was used as an opportunity to learn the important basics of data engineering - I found it to be really helpful in developing a base-level understanding!

**Comment by Dhruv-0987** [comment]
The data series was amazing, I believe because of the core functional programming approach everyone has nobody focusses on the data, which is essentially the most important part. Its easy to solve a problem in the code but to realize that there is something missing in the database schema is a nightmare, this tells us to focus equally on the data compared to the code.

**Comment by jontygardner** [comment]
Companies that know how to manage, analyse, and sort through their data are what separates them from everyone else. That's why I suggested that everyone watch Moneyball - data alone can help companies succeed. 

For non-devs, everyone should be skilled in at least data analytics and the best tools to draw out the data needed. Keen for an SSW data engineer to join the team!

**Comment by JeanThirion** [comment]
It's an important topic, it's in pretty much all projects, and data is usually the least known "tier" in our projects (for the average dev). 
Without a decent understanding of data, it's hard to design and impossible to architect a good solution for clients. I mean you can do good, but you can't do great.

**Comment by sethdaily** [comment]
My understanding or big data went from a 0 to somewhere between 2 and 4 I think. I liked the series because each week built on top of the previous one, and the last one put everything together nicely. Great learning experience for me, and no doubt a topic that will only become more important as data becomes more and more accessible.

**Comment by MattParkerDev** [comment]
I loved this video! It's awesome to get some feedback in the form of Digesting the Fat on our CTF submissions, and highlights the value of our CTFs :)

**Comment by AlexRothwell** [comment]
It was great to see the feedback loop being closed on this! 

I've got some suggestions for future series:
1. With an ideal timebox of 15 min, I think if we're intending on a series anyway, it'd be good to cut some weeks down a bit to better keep within that time
2. While it was great to see the responses from the series being reviewed, I think it would be more valuable to spend time presenting insights gained from the responses and any potential actions that are coming out of it. Relaying the response summaries which are already available to those interested does not feel like an effective use of time

**Comment by william-liebenberg** [comment]
Its great that we managed to cover such a big topic over the whole team and be able to have conversations about it.

**Comment by jimmidier** [comment]
I like the idea of making a CTF series with a retro episode at the end. It really helps a lot putting together all the knowledge contributed by all devs and share across. A huge thanks for people who made the efforts.

**Comment by jerryluossw** [comment]
I really like this series, data is the core asset of a business, how to make better use of data is one of the core competencies of future enterprises.
In China, more and more clients are aware of the role of data, so when they started the project, they will clarify the role of data at first.

**Comment by babakamyljanovssw** [comment]
This was great and interesting series. I've learned about datalakes, datawarehouse, and some tools. I'd love to see this kind of series again.

**Comment by jomelsontanco** [comment]
### Firstly, sign me up on data-related engagements please! :)

### Feedback

This month's digesting the fat gave me an interesting insight about how data works. 

Data is created out of data out of data in infinite feedback loops.

Take this week's video featuring Adam and Piers as an example. To put it in data terms, it's data analysis about folks at SSW's data analysis of videos about data. 

On top of that, we have people's feedbacks about this week's video itself, which is technically a data analysis about a data analaysis video about what folks at SSW's data analyses about data videos.

**Comment by tino-liu** [comment]
This series is really eye-widening. It doesn't just lightly brush over a huge topic but actually goes in with detailed explanation. I think that's what really contributed to stimulating people's eagerness to learn about Data Engineering.

**Comment by jernejk** [comment]
I really liked the retro on CTF.
The series idea is excellent, but CTFs are currently a "fire-and-forget" mechanism with no follow-up. It's easier to remember what I did 1-4 weeks ago with a good summary and what I thought about it.

The video does help, though. But trying to listen to it at 2x speed, so I can get everything done before the next meeting while also trying to unblock people on various things, it's a challenge. But that's mostly a "me" problem. 😌

**Comment by AnkyChe** [comment]
Love the retro, there is some fantastic feedback. As a non-developer, I've learned some things about data too.

**Comment by Aibono1225** [comment]
Awesome series and retro. I've learned so much during this whole journey. It's great to spend more weeks focusing on a big topic because we can dig into more details about it.

**Comment by johnxu16** [comment]
Having series and retro is a brilliant idea for CTF. However, the size of the series really affects the effects and experience of what would be left after a big series. As per a complex and large topic like data engineering, it would be much better and more memorable for me if we can separate it into two series (shorter).

**Comment by SofieHong** [comment]
Absolutely loved the retro on CTF ❤️
It's fascinating to compare feedback from four different CTFs. As a non-developer, I've gained so much insight from this unfamiliar field and I'm eager for more CTF series in the future. 🤩

**Comment by drwharris** [comment]
Great to see the learnings presented. Surprised me to see the broader interest in the topic.

**Comment by BrookJeynes** [comment]
I liked the retro. seeing how the answers changed as people learnt more was interesting. +1 to having bigger topics split into multiple CTF forms.

**Comment by ozair-ashfaque** [comment]
I got familiarized with the data engineering concepts and now I want to explore this field as well.

**Comment by matt-goldman** [comment]
Interesting series and it was cool to learn about. But it's not a topic that particularly interests me.

**Comment by Freego1783** [comment]
It's nice having a retro on CTF !

I've learned a lot of things in this series, but I don't see myself becoming a Data Engineer

**Comment by chrisschultzssw** [comment]
I like having a retro at the end of a series!
This was an important topic, since it sounds like we need more people with data skills.

**Comment by danielmackay** [comment]
Great wrap of up the series.  I watched the whole video and it was interesting to see everyone's feedback.

On the topic of proposed Data Warehouse / Data Lake ideas, anything related to commits or PBIs would be farmed as the size of commits or PBIs would be adjusted to suit.  If we related this to PBI Story Points though, this may help cut through this somewhat.

**Comment by Levijj22** [comment]
Love Digesting the Fat! Perfect way to round off the topic. 
It's great to see so many devs showing interest in data. 
I think that the further we progress technologically, the more data we will have to manage, especially with the influx of AI. 
We should really be concentrating on those who have shown interest and encourage them to hone their skills. I have a feeling we will be receiving a lot of data enquiries over the next 12 months!

**Comment by SamBWagner** [comment]
Wrap ups are super great and I really like this one! However, it was just too long to be watched in full while on client work
